src/mpi/group/group_impl.c

```diff
@@ -622,3 +622,47 @@ int MPIR_Group_union_impl(MPIR_Group * group_ptr1, MPIR_Group * group_ptr2,
   fn_fail:
     goto fn_exit;
 }
+
+int MPIR_Group_from_session_pset_impl(MPIR_Session * session_ptr, const char *pset_name,
+                                      MPIR_Group ** new_group_ptr)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_Group *group_ptr;
+
+    if (strcmp(pset_name, "mpi://WORLD") == 0) {
+        mpi_errno = MPIR_Group_create(MPIR_Process.size, &group_ptr);
```

Comment:
Is it possible to reuse `MPIR_Comm_group_impl` for `MPIR_Group_from_session_pset_impl`?

Reply:
Most of the session API is local function that can't assume a communicator exist.

Reply:
> When this function is called, I assume `MPIR_Process.comm_world` and `MPIR_Process.comm_self` are already established.

No, this assumption is false, so the checking of whether `comm_world` is established is critical. 

Now, in this PR, due to doing global init at `MPI_Session_init`, we always have `comm_world` active, But the code here cannot use that assumption. As a matter of fact, the next PR right after this will make the assumption no longer always true. The next PR will replace the assertions in the other branches with proper code as well.

The reason we do this PR separately from the next PR (proper local session init) is because this PR is still backward compatible; the next PR will have higher impact to downstream, thus better to separate.

Reply:
> Besides,`MPIR_Comm_group_impl` is also a local function,

`MPIR_Comm_group_impl` is not a local function.